CRM/Export/BAO/ExportProcessor.php

```diff
@@ -253,6 +253,29 @@ public function getRelationshipValue($relationshipType, $contactID, $field) {
     return isset($this->relatedContactValues[$relationshipType][$contactID][$field]) ? $this->relatedContactValues[$relationshipType][$contactID][$field] : '';
   }
 
+  /**
+   * Store the household Id if it has relationships.
+   * This is used by the merge same households to prevent duplicate households on export
+   * If a household has no relationships (it is not in this list) we export explicitly,
+   *   otherwise we rely on implicit export via "merge same household".
+   *
+   * @param int $householdId
+   */
+  public function setHouseholdHasRelationships($householdId) {
+    $this->relatedContactValues['householdRelationships'][$householdId] = TRUE;
+  }
+
+  /**
+   * If a household has relationships return TRUE, else return FALSE.
+   *
+   * @param $householdId
+   *
+   * @return bool
+   */
+  public function getHouseholdHasRelationships($householdId) {
+    return isset($this->relatedContactValues['householdRelationships'][$householdId]);
+  }
+
```

Comment:
This adds a store for all householdIds that have relationships.  We use this to check if a household should be directly exported or if it will be exported via the relationships.  As contact records will come in any order we don't know if household or individual will be processed first.